### ISSUE TITLE
Fix mypy-ribasim-qgis and add to CI

### ISSUE THREAD

**Issue by visr** [Deltares/Ribasim#1067]
This would need this commit:
[Add mypy to python_lint github actions](https://github.com/Deltares/Ribasim/pull/1065/commits/8ab4a9076303f1d941adbde158c936264cf380fc)

And the fix to these issues that it finds:
```
✨ Pixi task (default): mypy ribasim_qgis
Warning: --strict-concatenate is deprecated; use --extra-checks instead
ribasim_qgis/tests/test_load_plugin.py:[5](https://github.com/Deltares/Ribasim/actions/runs/7757356113/job/21156633632#step:8:6): error: Class cannot subclass "TestCase" (has type "Any")  [misc]
ribasim_qgis/tests/test_load_plugin.py:12: error: Class cannot subclass "TestCase" (has type "Any")  [misc]
ribasim_qgis/ribasim_qgis.py:[10](https://github.com/Deltares/Ribasim/actions/runs/7757356113/job/21156633632#step:8:11): error: Class cannot subclass "QgsDockWidget" (has type "Any")  [misc]
ribasim_qgis/tests/core/test_model.py:11: error: Class cannot subclass "TestCase" (has type "Any")  [misc]
Found 4 errors in 3 files (checked 22 source files)
```

**Comment by deltamarnix** [comment]
@Hofer-Julian We found out that github actions is using pixi 0.13.0, and that makes it fail. When we revert to 0.12.0 everything is fine and we have no mypy issues.

It seems like pixi 0.13.0 is breaking our environment for mypy.
We have tried installing 0.13.0 and threw away our lock file so 0.13.0 could make a new one. But that didn't seem to resolve the issue. How could we make a simple environment and report an issue to the pixi developers?
